File: kvx_flutter/lib/domain/entities/relay_statistics.dart
class RelayStatistics {
  final String deviceId;
  final int relayIndex;
  final Duration totalOnTime;
  final int activationCount;
  final DateTime? lastActivated;

  const RelayStatistics({
    required this.deviceId,
    required this.relayIndex,
    required this.totalOnTime,
    required this.activationCount,
    this.lastActivated,
  });

  RelayStatistics copyWith({
    String? deviceId,
    int? relayIndex,
    Duration? totalOnTime,
    int? activationCount,
    DateTime? lastActivated,
  }) {
    return RelayStatistics(
      deviceId: deviceId ?? this.deviceId,
      relayIndex: relayIndex ?? this.relayIndex,
      totalOnTime: totalOnTime ?? this.totalOnTime,
      activationCount: activationCount ?? this.activationCount,
      lastActivated: lastActivated ?? this.lastActivated,
    );
  }

  @override
  bool operator ==(Object other) =>
      identical(this, other) ||
      other is RelayStatistics &&
          runtimeType == other.runtimeType &&
          deviceId == other.deviceId &&
          relayIndex == other.relayIndex;

  @override
  int get hashCode => Object.hash(deviceId, relayIndex);
}

File: kvx/Models/RelayStatistics.swift
//
//  RelayStatistics.swift
//  kvx
//
//  Created by Developer Agent on 06/09/2026.
//

import Foundation

struct RelayStatistics: Equatable, Hashable, Codable {
    let deviceId: String
    let relayIndex: Int
    let totalOnTime: TimeInterval  // seconds
    let activationCount: Int
    let lastActivated: Date?

    init(
        deviceId: String,
        relayIndex: Int,
        totalOnTime: TimeInterval,
        activationCount: Int,
        lastActivated: Date? = nil
    ) {
        self.deviceId = deviceId
        self.relayIndex = relayIndex
        self.totalOnTime = totalOnTime
        self.activationCount = activationCount
        self.lastActivated = lastActivated
    }

    // Equatable based on deviceId and relayIndex
    static func == (lhs: RelayStatistics, rhs: RelayStatistics) -> Bool {
        lhs.deviceId == rhs.deviceId && lhs.relayIndex == rhs.relayIndex
    }

    // Hashable based on deviceId and relayIndex
    func hash(into hasher: inout Hasher) {
        hasher.combine(deviceId)
        hasher.combine(relayIndex)
    }
}
